nnef/src/registry.rs

```diff
@@ -217,20 +217,29 @@ impl Registry {
 
             // mitigation of nnef "scalar" type mismatch with tract-core more
             // strict types
-            if a_dt != b_dt {
+            if (!a_dt.is_quantized() || !b_dt.is_quantized()) && a_dt != b_dt {
                 if builder.model.node(a.node).op_is::<tract_core::ops::konst::Const>() {
                     a = builder.wire_as_outlets(tract_core::ops::cast::cast(b_dt), &[a])?[0];
                 } else {
                     b = builder.wire_as_outlets(tract_core::ops::cast::cast(a_dt), &[b])?[0];
                 };
             }
             let inputs = multicast(builder, &[a, b])?;
-            let mut wire = builder
-                .wire_as_outlets(tract_core::ops::binary::TypedBinOp(bin.1.clone()), &inputs)?[0];
-            if let Some(Some(out_dt)) = dt.first() {
-                if out_dt != &a_dt {
-                    wire =
-                        builder.wire_as_outlets(tract_core::ops::cast::cast(*out_dt), &[wire])?[0];
+
+            let c_dt: Option<DatumType> = match dt.first().cloned() {
+                Some(c_dt) => c_dt,
+                None => None,
+            };
```

Comment:
Nitpicking: I think there's a more idiomatic way :) with `then` or `and then` maybe ?